builtin/v9/migration/verifreg.go

```diff
@@ -99,14 +76,19 @@ func migrateVerifreg(ctx context.Context, adtStore adt8.Store, priorEpoch abi.Ch
 			allocationsMapMap[clientIDAddress] = clientAllocationMap
 		}
 
+		expiration := verifreg9.MaximumVerifiedAllocationExpiration + priorEpoch
+		if expiration > proposal.StartEpoch {
+			expiration = proposal.StartEpoch
+		}
+
 		if err = clientAllocationMap.Put(nextAllocationID, &verifreg9.Allocation{
 			Client:     abi.ActorID(clientIDu64),
 			Provider:   abi.ActorID(providerIDu64),
 			Data:       proposal.PieceCID,
 			Size:       proposal.PieceSize,
 			TermMin:    proposal.Duration(),
-			TermMax:    market9.DealMaxDuration,
-			Expiration: verifreg9.MaximumVerifiedAllocationExpiration + priorEpoch,
+			TermMax:    market9.DealMaxDuration + market9.MarketDefaultAllocationTermBuffer,
```

Comment:
I think we want `proposal.Duration() + market9.MarketDefaultAllocationTermBuffer`